homeassistant/components/homematicip_cloud/climate.py

```diff
@@ -260,23 +266,32 @@ def _device_profiles(self) -> list[Any]:
         return [
             profile
             for profile in self._device.profiles
-            if profile.visible
-            and profile.name != ""
-            and profile.index in self._relevant_profile_group
+            if profile.visible and profile.index in self._relevant_profile_group
         ]
 
     @property
     def _device_profile_names(self) -> list[str]:
         """Return a collection of profile names."""
-        return [profile.name for profile in self._device_profiles]
+        return [
+            self._get_qualified_profile_name(profile)
+            for profile in self._device_profiles
+        ]
+
+    def _get_qualified_profile_name(self, profile) -> str:
```

Comment:
Please add a docstring explaining what the function does

Reply:
Done